rig-core/src/providers/huggingface/client.rs

```
@@ -38,11 +41,24 @@ impl SubProvider {
     /// in the url and in the request body.
     pub fn transcription_endpoint(&self, model: &str) -> String {
         match self {
-            SubProvider::HFInference => format!("hf-inference/models/{}", model),
+            SubProvider::HFInference => format!("/{}", model),
             _ => panic!("transcription endpoint is not supported yet for {}", self),
         }
     }
 
+    /// Get the image generation endpoint for the SubProvider
+    /// Required because Huggingface Inference requires the model
+    /// in the url and in the request body.
+    pub fn image_generation_endpoint(&self, model: &str) -> String {
+        match self {
+            SubProvider::HFInference => format!("/{}", model),
+            _ => panic!(
```

I am not sure we should panic here, we should return an error

---

Fair enough, I swapped it to a Result